synapse/lib/stormlib/aha.py

```diff
@@ -531,5 +656,226 @@ async def _methPoolSvcDel(self, svcname):
+                )
+                $lib.print($printer.row($row))
+            }
+        }
+
+        $virtual_services = ({})
+        $member_servers = ({})
+
+        for $svc in $lib.aha.list() {
+            $name = $svc.name
+            $svcinfo = $svc.svcinfo
+            $urlinfo = $svcinfo.urlinfo
+            $hostname = $urlinfo.hostname
+
+            if ($name != $hostname) {
+                $virtual_services.$name = $svc
+            } else {
+                $member_servers.$name = $svc
+            }
+        }
+
+        $mirror_groups = ({})
+        for ($vname, $vsvc) in $virtual_services {
+            $vsvc_info = $vsvc.svcinfo
+            $vsvc_iden = $vsvc_info.iden
+            $vsvc_leader = $vsvc_info.leader
+            $vsvc_hostname = $vsvc_info.urlinfo.hostname
+
+            if (not $vsvc_iden or not $vsvc_hostname or not $vsvc_leader) {
+                continue
+            }
+
+            $primary_member = $member_servers.$vsvc_hostname
+            if (not $primary_member) {
+                continue
+            }
+
+            $members = $lib.list($primary_member)
+            for ($mname, $msvc) in $member_servers {
+                if ($mname != $vsvc_hostname) {
+                    $msvc_info = $msvc.svcinfo
+                    if ($msvc_info.iden = $vsvc_iden and $msvc_info.leader = $vsvc_leader) {
+                        $members.append($msvc)
+                    }
+                }
+            }
+
+            if ($lib.len($members) > 1) {
+                $mirror_groups.$vname = $members
+            }
+        }
+
+        for ($vname, $members) in $mirror_groups {
+            $cell_infos = $get_cell_infos($vname)
+            $group_status = $build_status_list($members, $cell_infos)
+            $lib.print('Service Mirror Groups:')
+            $output_status($vname, $group_status, $printer)
+
+            if $check_sync_status($group_status) {
+                $lib.print('Group Status: In Sync')
+            } else {
+                $lib.print(`Group Status: Out of Sync; Waiting {$timeout} seconds for sync...`)
+                $leader_nexs = $lib.null
+                for $status in $group_status {
+                    if (($status.role = 'leader') and ($status.nexs_indx > 0)) {
+                        $leader_nexs = $status.nexs_indx
+                        break
+                    }
+                }
+
+                if $leader_nexs {
+                    $start_time = $lib.time.now()
+
+                    while $lib.true {
+                        $now = $lib.time.now()
+                        $elapsed = ($now - $start_time)
+                        if ($elapsed >= ($timeout * 1000)) {
+                            $lib.raise(TimeOut, mesg=`Mirror sync timeout after {$timeout} seconds`)
+                        }
+
+                        $todo_timeout = $lib.min(5, ((($timeout * 1000) - $elapsed) / 1000))
+                        $responses = $lib.list()
```

Comment:
Can we replace all the `$lib.list()` use in this PR with built-in list syntax?

ex.

```
$response = ()
```

Reply:
I didn't know about this shortcut for list, thanks will make the change in the next commit!